packages/provider/index.js

```diff
@@ -30,6 +32,39 @@ module.exports = {
         { keepAlive: false }
       );
     }
+    if (
+      typeof provider.request === "undefined" &&
+      typeof provider.send !== "undefined"
+    ) {
```

Comment:
Instead of doing this here, I think it might make more sense to do this for `request` near the same place we wrap `send` (https://github.com/trufflesuite/truffle/blob/develop/packages/provider/wrapper.js#L26-L34)

This way all the wrapping logic is cohesive to that `wrapper.js` module.

Unless you think that this is a better place to handle EIP-1193 separately? I'm not sure.

Reply:
I, kind of agree with you @gnidan. `wrapper.js` might make more sense to handle EIP-1193 wrapping. Any thoughts on this @cds-amal @davidmurdoch @eggplantzzz @lsqproduction ?

Reply:
I don't have a preference.

Reply:
Should we make a separate PR to handle the implementation of EIP-1193?

Reply:
I have moved the EIP-1193 wrapping to `wrapper.js` as suggested by @gnidan. I thought of keeping this implementation in this PR itself, since it is related to `truffle call` command.

Reply:
Oh sorry, I missed your question. I think a separate PR for the provider stuff would be good because it'll likely get more eyes that way.

Reply:
I will make a separate PR for this. Thanks for clarifying, @gnidan. :)